src/modules/involvementAPI.js

```diff
@@ -0,0 +1,94 @@
+import { Comment, Reserve } from './Modal.js';
+
+export default class InvolvementAPI {
+  static commentsURL =
+    'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/1t7BiC343viu9RJP3IIx/comments'
+
+  static likesURL =
+    'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/1t7BiC343viu9RJP3IIx/likes'
+
+  static reserveURL =
+    'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/1t7BiC343viu9RJP3IIx/reservations'
+
+  static fetchComments = async (launchId) => {
+    const comments = [];
+    await fetch(`${InvolvementAPI.commentsURL}?item_id=${launchId}`)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((json) => {
+        json.forEach((c) => {
+          comments.push(new Comment(c.username, c.creation_date, c.comment));
+        });
+      });
+    return comments;
+  }
+
+  static postComment = async (launchId, newcomment) => {
+    await fetch(InvolvementAPI.commentsURL, {
+      method: 'POST',
+      body: JSON.stringify({
+        item_id: launchId,
+        username: newcomment.username,
+        comment: newcomment.comment,
+      }),
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+  }
+
+  static fetchAllLikes = async () => fetch(InvolvementAPI.likesURL).then((response) => {
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    return response.json();
+  })
+
+  static postLike = async (launchId) => {
+    await fetch(InvolvementAPI.likesURL, {
+      method: 'POST',
+      body: JSON.stringify({
+        item_id: launchId,
+      }),
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+  }
+
+  static getReservation = async (launchId) => {
+    const reserves = [];
+    await fetch(`${InvolvementAPI.reserveURL}?item_id=${launchId}`)
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        return response.json();
+      })
+      .then((json) => {
+        json.forEach((c) => {
+          reserves.push(new Reserve(c.username, c.date_start, c.date_end));
+        });
+      });
+    return reserves;
+  }
+
+  static postReservation = async (launchId, newreservation) => {
+    await fetch(InvolvementAPI.reserveURL, {
+      method: 'POST',
+      body: JSON.stringify({
+        item_id: launchId,
+        username: newreservation.username,
+        date_start: newreservation.date_start,
+        date_end: newreservation.date_end,
+      }),
+      headers: {
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+  }
+}
```

Comment:
- [ OPTIONAL ] Great work on implementing the reservations feature, especially considering it was an optional requirement! Adding this functionality showcases your dedication and willingness to go the extra mile. 🔥 

- However, I noticed that when the reservation button is clicked, it throws an error. To ensure the feature works smoothly and meets expectations, I recommend addressing this issue. Take the time to debug and refine the functionality, ensuring it is error-free and provides a seamless user experience. 😉 

- Keep up the good work and continue refining your skills. I'm confident that with a bit of attention to detail, your reservations feature will be a valuable addition to the project. Well done! 👏 
- Error Message
<img width="1280" alt="Screen Shot 2023-05-25 at 3 47 01 PM" src="https://github.com/unorjikingsley/Rocket-Launch/assets/102699958/5c35c8cb-ed06-4532-a7b6-67d95a398f71">